### DISCUSSION TITLE
Tags tracking in sidebar

### DISCUSSION THREAD

**Discussion by asshuka** [AprilSylph/XKit-Rewritten#100] [Help Desk] [answered]
(I would like to apologize if I labelled this incorrectly or otherwise cause some mistakes, this is the first time I've used GitHub and am not aware of my way around or the general etiquettes, though I'm following the guidelines the best I can >_>)
I was directed here to ask more about the tracked tags in sidebar part of Tags Tracking+. I noticed that it was marked as done in the progress board, but crossed out. I'm wondering if it wasn't or couldn't be implemented due to complications, and/or if there's any possible way to implement it. Thank you!

**Comment by AprilSylph** [comment, marked as answer]
Tag Tracking+ originally did put tracked tags in the sidebar with an unread count, but with the way the new Tumblr dashboard works, it introduced the question of how often to refresh the unread counts. However, there is no one-size-fits-all answer to that, and building a preference for it could allow users to spam the Tumblr API, which may have consequences for their account(s).

Given this, and the fact that I was building unread counts into the Tumblr search dropdown anyway, I decided the best answer to this problem was to remove Tag Tracking+'s sidebar section entirely, so that unread counts would only be fetched upon user input. This also solves the problem of needing to construct links rather than just tack a number onto existing ones, and allows visiting tag pages to consistently use Tumblr's soft-page-switch mechanism.

And lastly, XKit Rewritten is designed to also target the tablet and mobile layouts of the dashboard. Building or continuing features which only work on the desktop layout is something I'm avoiding where there are alternative solutions which work across all layouts, to keep the experience consistent. (This rationale may seem odd as Firefox for Android doesn't currently support the entire range of addons on AMO, but Firefox for Android Nightly does experimentally [allow any addon from AMO to be installed](https://blog.mozilla.org/addons/2020/09/29/expanded-extension-support-in-firefox-for-android-nightly/), so it can be assumed that support for all addons will be making a comeback in the future.)
